test/core/test_checks.py

```diff
@@ -0,0 +1,201 @@
+    (chk_equal, [[1, 2, 3, 4], [1, 2, 3, 4, 5]], {}),
+    (chk_equal, [{"a": 1, "b": 2}, {"b": 1, "a": 2}], {}),
+    (chk_equal, [2 + 1j, 1 + 1j], {}),
+]
+
+
+@pytest.mark.parametrize(("check", "args", "kwargs"), chk_fail_data)
+def test_fixate_checks_fail(check: Callable, args, kwargs):
+    """test checks fail"""
+    with pytest.raises(CheckFail):
+        check(*args, **kwargs)
+
+
+@pytest.fixture
+def mock_check_string(monkeypatch):
+    """Mock the call to sequencer.check to return the check_string"""
+
+    def mock_check(self, chkresult: CheckResult):
+        return chkresult.check_string
+
+    monkeypatch.setattr(fixate.sequencer.Sequencer, "check", mock_check)
+
+
+chk_ui_string = [
+    (chk_passes, [], {"description": "test"}, "PASS: test"),
+    (
+        chk_log_value,
+        [123],
+        {"description": "test float .2", "fmt": ".2f"},
+        f"PASS: {123:.2f} : test float .2",
+    ),
+    (
+        chk_log_value,
+        [123.456],
+        {"description": "test float", "fmt": ".2f"},
+        f"PASS: {123.456:.2f} : test float",
+    ),
+    # (chk_log_value, [123.456], {"fmt": ".2f"}, ""),
+    # (chk_log_value, [123], {"fmt": ".2f"}, ""),
+    # (chk_equal, ["string", "string"], {}),
+    # (chk_equal, [1e5, 1e5], {}),
+    # (chk_equal, [[1,2,3,4], [1,2,3,4]], {}),
+    # (chk_equal, [(1,2,3,4), (1,2,3,4)], {}),
+    # (chk_equal, [{"a":1, "b":2}, {"a":1, "b":2}], {}),
+    # (chk_equal, [2+1j, 2+1j], {}),
+]
+
+
+@pytest.mark.parametrize(("check", "args", "kwargs", "check_string"), chk_ui_string)
+def test_checks_formatting(
+    mock_check_string, check: Callable, args, kwargs, check_string
+):
+    """Test different formatting options"""
+    assert check(*args, **kwargs) == check_string
+
+
+def test_checks_logging():
+    """TODO: somehow test checks are logged properly"""
+    # NOTE: tiny coverage in test_script2log
+    pass
+
+
+test_raise_data = [
+    # Missing args - probably overkill
+    (chk_in_range, [], {}, TypeError),
+    (chk_in_tolerance, [], {}, TypeError),
+    (chk_in_range_equal, [], {}, TypeError),
+    (chk_outside_range, [], {}, TypeError),
+    (chk_outside_range_equal, [], {}, TypeError),
+    (chk_outside_range_equal_min, [], {}, TypeError),
+    (chk_outside_range_equal_max, [], {}, TypeError),
+    (chk_smaller_or_equal, [], {}, TypeError),
+    (chk_greater_or_equal, [], {}, TypeError),
+    (chk_smaller, [], {}, TypeError),
+    (chk_greater, [], {}, TypeError),
+    (chk_true, [], {}, TypeError),
+    (chk_false, [], {}, TypeError),
+    (chk_in_tolerance_equal, [], {}, TypeError),
+    (chk_in_deviation_equal, [], {}, TypeError),
+    (chk_equal, [], {}, TypeError),
+    # Invalid args
+    (chk_greater, [1, "test"], {}, TypeError),
+    # Format exceptions are swallowed so hard to test
+]
+
+
+@pytest.mark.parametrize(("check", "args", "kwargs", "exception"), test_raise_data)
+def test_checks_raise(check: Callable, args, kwargs, exception: Exception):
```

Comment:
Typehinting should be type[Exception] as you're giving it an exception type, not an actual instance of an Exception

Reply:
Changed to Type[Exception]